.github/workflows/push-tag-release.yml

```diff
@@ -1,69 +1,263 @@
+          echo "VERSION=$(echo ${{ github.ref }} | cut -d/ -f 4-4)" >> $GITHUB_ENV
+          echo "TAG=$(echo ${{ github.ref }} | cut -d/ -f 3-4)" >> $GITHUB_ENV
+      - name: set apps folder
+        if: ${{ env.PACKAGE == 'connect' }}
+        shell: bash
+        run: |
+          echo "FOLDER=apps/${{ env.PACKAGE }}" >> $GITHUB_ENV
+          echo "TO_PYPI=false" >> $GITHUB_ENV
+      - name: set installers folder
+        if: ${{ env.PACKAGE == 'connect-installer' }}
+        run: echo Not supported by this CI && exit 1
+      - name: set lib folder
+        if: ${{ env.PACKAGE == 'constants' || env.PACKAGE == 'framework-core' || env.PACKAGE == 'framework-qt' || env.PACKAGE == 'qt' || env.PACKAGE == 'qt-style' || env.PACKAGE == 'utils' }}
+        shell: bash
+        run: |
+          echo "FOLDER=libs/${{ env.PACKAGE }}" >> $GITHUB_ENV
+          echo "TO_PYPI=true" >> $GITHUB_ENV
+      - name: set projects folder
+        if: ${{ env.PACKAGE == 'application-launcher' || env.PACKAGE == 'connect-action-launcher-widget' || env.PACKAGE == 'connect-plugin-manager' || env.PACKAGE == 'connect-publisher-widget' || env.PACKAGE == 'connect-timetracker-widget' || env.PACKAGE == 'framework-photoshop' || env.PACKAGE == 'nuke-studio' || env.PACKAGE == 'rv' }}
+        shell: bash
+        run: |
+          echo "FOLDER=projects/${{ env.PACKAGE }}" >> $GITHUB_ENV
+          echo "TO_PYPI=false" >> $GITHUB_ENV
+      - name: debug
+        shell: bash
+        run: |
+          echo 'TO_PYPI: ${{ env.TO_PYPI }}'
+          echo 'FOLDER: ${{ env.FOLDER }}'
+          echo 'PACKAGE: ${{ env.PACKAGE }}'
+          echo 'VERSION: ${{ env.VERSION }}'
+          echo 'TAG: ${{ env.TAG }}'
+      - name: Failsafe delete previous releases, matching tag pattern
+        shell: bash
+        run: |
+          gh release list --limit 100 | sed 's/|/ /' | awk '{print $1, $8}' | grep ${{ env.TAG }} | while read -r line; do echo Deleting conflicting release: $line; gh release delete -y "$line"; done
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        continue-on-error: true
+      - name: Create release
+        shell: bash
+        run: |
+          gh release create ${{ env.TAG }} --title ${{ env.TAG }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-  # Release the package
-  release:
-    needs: pants-build
-    uses: ./.github/workflows/release.yml
-    with:
-      changed: ${{needs.pants-build.outputs.changed}}
-      pre: false
-      version: ${{ needs.pants-build.outputs.version }}
-      deploy_aws: true
-    secrets:
-      aws_key_id: ${{ secrets.S3_ACCESS_KEY }}
-      aws_secret_access_key: ${{ secrets.S3_SECRET_ACCESS_KEY }}
+  build-linux:
+    needs: set-variables-create-release
+    concurrency:
+      group: ci-${{ github.ref }}
+    if: ${{ needs.set-variables-create-release.outputs.package != 'framework-photoshop' }}
```

Comment:
I think we should by default always do the following steps (Which if I understood it correctly is almost what you already do):
- checkout
- setup poetry
- poetry_build
- build_connect_package
* if there is any special package that has to follow another setup like rv, we add an if for the rv and we build the rv-plugin.

Then theoretically we shouldn't be in need of building the style when want to release PS right? It should only be a requirement when releasing the qt-style library, And that could be an if somewhere. 
But in any case if that is needed when releasing any framework package or we have to release a different package depending on the OS, then I think we should do it with a matrix like we did in the old release.yml file where we specify the OS and tag the package with a suffix, so we don't have to duplicate the code.

Reply:
The CI is designed to build all plugins on Linux that are platform independent, for example Maya in the future.

This if gate is simply because Photoshop is not supported on Linux, and is a special case. IMO it is fine for now to have these if gates, instead of complicating the CI with matrixes that are hard to interpret and also have limitations - can recall you cannot drive the OS it runs on through matrix variables.

Reply:
Ok, But yes, you can drive the OS in the matrix as well, you have an example in the removed release.yaml file:

```
    strategy:
      matrix:
        # Add new packages here
        package: [application-launcher, connect-action-launcher-widget, connect-plugin-manager, framework-core, framework-qt, framework-maya, framework-nuke, framework-houdini, framework-3dsmax, connect-timetracker-widget, nuke-studio, rv]
        os: [ubuntu-latest]
        suffix: ['']
        include:
          - package: framework-unreal
            os: windows-latest
            suffix: 'win'
          # Add platform dependent packages here
```